domains/certificates/Query.API/API.IntegrationTests/Repositories/SynchronizationMigrationTests.cs

```diff
@@ -0,0 +1,137 @@
+    {
+        var gsrn = GsrnHelper.GenerateRandom();
+        var sync1 = new SyncPosition(Guid.NewGuid(), gsrn, 42);
+        var sync2 = new SyncPosition(Guid.NewGuid(), gsrn, 43);
+        var sync3 = new SyncPosition(Guid.NewGuid(), gsrn, 44);
+
+        await SaveSyncPositions(sync1, sync2, sync3);
+
+        await store.MigrateSynchronizationPosition(CancellationToken.None);
+
+        await using var session = store.QuerySession();
+
+        var syncPositions = await session.Query<SyncPosition>().ToListAsync();
+        syncPositions.Should().HaveCount(0);
+
+        var synchronizationPositions = await session.Query<SynchronizationPosition>().ToListAsync();
+        synchronizationPositions.Should().HaveCount(1);
+        synchronizationPositions.Single().Should().BeEquivalentTo(new SynchronizationPosition { GSRN = gsrn, SyncedTo = 44 });
+    }
+
+    [Fact]
+    public async Task can_be_called_multiple_times()
+    {
+        var gsrn = GsrnHelper.GenerateRandom();
+        var sync1 = new SyncPosition(Guid.NewGuid(), gsrn, 42);
+
+        await SaveSyncPositions(sync1);
+
+        // Call multiple times
+        await store.MigrateSynchronizationPosition(CancellationToken.None);
+        await store.MigrateSynchronizationPosition(CancellationToken.None);
+        await store.MigrateSynchronizationPosition(CancellationToken.None);
+
+        await using var session = store.QuerySession();
+
+        var syncPositions = await session.Query<SyncPosition>().ToListAsync();
+        syncPositions.Should().HaveCount(0);
+
+        var synchronizationPositions = await session.Query<SynchronizationPosition>().ToListAsync();
+        synchronizationPositions.Should().HaveCount(1);
+        synchronizationPositions.Single().Should().BeEquivalentTo(new SynchronizationPosition { GSRN = gsrn, SyncedTo = 42 });
+    }
+
+    [Fact]
+    public async Task migrates_20_metering_points_with_1_year_positions()
+    {
+        const int hoursInAYear = 24 * 365;
+        const int numMeteringPoints = 20;
+
+        var gsrns = Enumerable.Range(1, numMeteringPoints).Select(_ => GsrnHelper.GenerateRandom()).ToArray();
+
+        int i = 0;
+        foreach (var gsrn in gsrns)
+        {
+            var positionsForMeteringPoint = Enumerable.Range(1, hoursInAYear)
+                .Select(h => new SyncPosition(Guid.NewGuid(), gsrn, h + i))
+                .ToArray();
+
+            await SaveSyncPositions(positionsForMeteringPoint);
+            i++;
+        }
+
+        await store.MigrateSynchronizationPosition(CancellationToken.None);
+
+        await using var session = store.QuerySession();
+
+        var syncPositions = await session.Query<SyncPosition>().ToListAsync();
+        syncPositions.Should().HaveCount(0);
+
+        var synchronizationPositions = await session.Query<SynchronizationPosition>().ToListAsync();
+        var expected = gsrns.Select((gsrn, cnt) => new SynchronizationPosition { GSRN = gsrn, SyncedTo = hoursInAYear + cnt });
```

Comment:
Just by reading this test its hard for me to see whats is expected. Could you be clear in the variable name what it is?

Reply:
Sure. Which variable name is it, that should be renamed?